funsor/affine.py

```diff
@@ -57,6 +57,12 @@ def eager_subs(self, subs):
 def eager_affine(const, coeffs):
     if not coeffs:
         return const
+    if not all(isinstance(var, Variable) for var, coeff in coeffs):
```

Comment:
When would a `Affine(...)` be called with non-variables?

Reply:
This happens as a result of the changes to substitution, which reuses the interpreter and patterns here in this file to normalize the result of substitution.  Compare to the logic in `Affine.eager_subs`, where that re-normalization was hard-coded.